content/4_Gold/Combinatorics.mdx

```diff
@@ -146,19 +164,19 @@ int binomial(int n, int k, int p) {
 
 ### Method 2: Factorial Definition (Modular Inverses) - $\mathcal{O}(n\log MOD)$
 
-First of all, we define $n!$ as $n \times (n - 1) \times (n - 2) \times \ldots 1$, which represents the number of permutations of a set of $n$ elements.
+Define $n!$ as $n \times (n - 1) \times (n - 2) \times \ldots 1$. $n!$ represents the number of permutations of a set of $n$ elements. See [this AoPS Article](https://artofproblemsolving.com/wiki/index.php/Factorial) for more details.
 
 Another way to calculate binomial coefficients is as follows:
 
 $$
 	\binom{n}{k} = \frac{n!}{k!(n-k)!}
 $$
 
-Since there are $n!$ permutations of $n$ elements, we go through all permutations and always include the first $k$ elements of the permutation in the subset. The order of the elements inside and outside the subset does not matter, so the result is divided by $k!$ and $(n − k)!$
+Recall that $\binom{n}{k}$ also represents the number of ways to choose $k$ elements from a set of $n$ elements. One strategy to get all such combinations is to go through all possible permutations of the $n$ elements, and only pick the first $k$ elements out of each permutation. There are $n!$ ways to do so. However, note the the order of the elements inside and outside the subset does not matter, so the result is divided by $k!$ and $(n − k)!$
 
 Since these binomial coefficients are large, problems typically require us to output the answer modulo a large prime $p$ such as $10^9 + 7$.
 
-Fortunately, we can use [modular inverses](/gold/modular) to divide $n!$ by $k!$ and $(n - k)!$ modulo $p$. We can precompute all factorials in $\mathcal{O}(n)$ time and inverse factorials in $\mathcal{O}(n\log MOD)$ (although not necessary, you can also precompute inverse factorials in $\mathcal{O}(n)$ time).
+Fortunately, we can use [modular inverses](/gold/modular) to divide $n!$ by $k!$ and $(n - k)!$ modulo $p$. A naive implementation of this would take too long, as computing inverse factorials online can be very time costly. Instead, we can precompute all factorials in $\mathcal{O}(n)$ time and inverse factorials in $\mathcal{O}(n\log MOD)$ (although not necessary, you can also precompute inverse factorials in $\mathcal{O}(n)$ time). See the code below for the implementation.
```

Comment:
why do you use $p$ in the first sentence but $MOD$ in the third?

Reply:
I don't (I didn't write the editorial, but I'll edit that. And yes, $p$ and $MOD$ are different, and should remain different, becuase $p$ refers to any prime, as modular inverses work for any prime. The $MOD$ we use happens to also be prime (1e9+7), but it's not the same as $p$.